core/snmp/impl-snmp4j/src/main/java/org/opennms/netmgt/snmp/snmp4j/Snmp4JTrapNotifier.java

```diff
@@ -79,6 +81,10 @@ public Snmp4JV1TrapInformation(InetAddress agent, String community, PDUv1 pdu, T
             super(agent, community, trapProcessor);
             m_pdu = pdu;
         }
+        
+        public int getPartionKey() {
```

Comment:
We should generate the partition key without altering the message class. Please change the code to use a new Camel processor class to generate the partition key.